src/context/CartContext.js

```diff
@@ -0,0 +1,12 @@
+import React, { useState, createContext} from 'react'
+
+export const CartContext = createContext()
+
+export const CartProvider = props => {
+    const [cartItems, setCartItems] = useState([])
```

Comment:
I recommend using `useReducer` here instead of `useState`. Inside of the reducer, you will be able to handle all of the state update logic and not need to have the logic in your component files.